File: Csharp/SolidCourse/SRP/After.cs
public class MyUser
{
    public long Id { get; set; }
    public string Username { get; set; }
    public string Password { get; set; }
    public string Email { get; set; }
}

public class UserCreator
{
    private Logger _logger = new Logger();
    
    public void AddUser(string username) 
    {
        // Add user logic
        Console.WriteLine($"User {username} added.");
        var message = $"User {username} added.";
        _logger.LogAction(message);
    }
}

public class UserRemover
{
    private Logger _logger = new Logger();
    
    public void RemoveUser(string username) 
    {
        // Add user logic
        Console.WriteLine($"User {username} removed.");
        var message = $"User {username} removed.";
        _logger.LogAction(message);
    }
}

public class Logger
{
    public void LogAction(string message) 
    {
        Console.WriteLine($"Log: {message}");
    }
}

File: CPlusPlus/SRP/After.cpp
#include <iostream>
#include <string>

/////////// MyUser.cpp
class MyUser {
public:
    long Id;
    std::string Username;
    std::string Password;
    std::string Email;
};

/////////// Logger.cpp
class Logger {
public:
    void LogAction(const std::string& message) {
        std::cout << "Log: " << message << std::endl;
    }
};


/////////// UserCreator.cpp
class UserCreator {
    private:
        Logger& logger;  // reference to shared Logger
    
    public:
        UserCreator(Logger& loggerRef) : logger(loggerRef) {}
    
        void AddUser(const std::string& username) {
            std::cout << "User " << username << " added." << std::endl;
            logger.LogAction("User " + username + " added.");
        }
    };
    

/////////// UserRemover.cpp
class UserRemover {
    private:
        Logger& logger;
    
    public:
        UserRemover(Logger& loggerRef) : logger(loggerRef) {}
    
        void RemoveUser(const std::string& username) {
            std::cout << "User " << username << " removed." << std::endl;
            logger.LogAction("User " + username + " removed.");
        }
    };
    